user-manager/src/test/java/com/qpay/usermanager/services/impl/MerchantServiceImplTest.java

```diff
@@ -0,0 +1,180 @@
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+
+@ExtendWith(MockitoExtension.class)
+class MerchantServiceImplTest {
+    @InjectMocks
+    private  MerchantServiceImpl merchantService;
+
+    @Mock
+    private MerchantRepository merchantRepository;
+
+    @Mock
+    private MerchantMapper merchantMapper;
+
+    private final String EMAIL = "bob@gmail.com";
+
+    private final MerchantCreation MERCHANT_CREATION = new MerchantCreation("bob", EMAIL, "word", 1L);
+
+    @Test
+    void should_returnMerchants() {
+        //given
+        var merchant1 = MerchantEntity.builder()
+                .name("bob")
+                .email("cool_bob@gmail.com")
+                .password("1234")
+                .walletId(1L)
+                .build();
+
+        var merchant2 = MerchantEntity.builder()
+                .name("greg")
+                .email("geg@gmail.com")
+                .password("5678")
+                .walletId(2L)
+                .build();
+
+        List<MerchantEntity> list = List.of(merchant1, merchant2);
+        given(merchantRepository.findAll()).willReturn(list);
+
+        //when
+        var result = merchantService.getMerchants();
+
+        //then
+        assertThat(result).isEqualTo(list);
+    }
+
+    @Test
+    void should_returnMerchantById() {
+        //given
+        var merchant1 = MerchantEntity.builder()
+                .name("bob")
+                .email("cool_bob@gmail.com")
+                .password("1234")
+                .walletId(1L)
+                .build();
+
+        given(merchantRepository.findById(1L)).willReturn(Optional.of(merchant1));
+
+        //when
+        var result = merchantService.getMerchantById(1);
+
+        //then
+        assertThat(result).isEqualTo(merchant1);
+    }
+
+    @Test
+    void should_throwException_when_merchantByIdNotFound() {
+        //when
+        Throwable throwable = catchThrowable(() -> merchantService.getMerchantById(1));
+
+        //then
+        assertThat(throwable).isInstanceOf(NoMerchantFoundException.class);
+    }
+
+    @Test
+    void should_addMerchant() {
+        //given
+       var expectedMerchant = MerchantEntity.builder()
+               .name("bob")
+               .email(EMAIL)
+               .password("word")
+               .walletId(1L)
+               .build();
+
+       given(merchantRepository.existsByEmail(EMAIL)).willReturn(false);
+       given(merchantMapper.map(MERCHANT_CREATION)).willReturn(expectedMerchant);
+
+       //when
+       var result = merchantService.addMerchant(MERCHANT_CREATION);
+
+       //then
+        assertThat(result).isEqualTo(expectedMerchant);
+    }
+
+    @Test
+    void should_throwException_when_addMerchantEmailAlreadyExist() {
```

Comment:
```suggestion
    void should_throwEmailAlreadyExistsException_when_newMerchantEmailAlreadyExist() {
```

the same for all other cases.